assets/scripts/Origami.js

```diff
@@ -0,0 +1,157 @@
+   */
+  generateSVG() {
+    const svg = document.createElement('object');
+    svg.data = this.svgPath;
+    document.body.appendChild(svg);
+    this.currentSVG = svg;
+    this.#addClickListeners();
+  }
+
+  removeCurrentSVG() {
+    this.currentSVG.remove();
+    this.currentSVG = null;
+  }
+
+  #addClickListeners() {
+    if (this.currentSVG.data.endsWith('closed-new.svg')) {
+      this.activateClosedHandler();
+    } else if (this.currentSVG.data.endsWith('-nums.svg')) {
+      this.activateNumsHandler();
+    }
+  }
+  /**
+   * Function that converts ID of flap to its color.
+   * @param {string} - string that identifies which flap it is on the fortune teller
+   * @returns {number} - unknown if invalid input, or a string that represents color of flap
+   */
+  idToColor(flapID) {
+    return COLOR_BY_CLICK_REGION[flapID] || 'unknown';
+  }
+    
+  /**
+   * Function that converts ID of flap to its number specifically for SVGs with numbers.
+   * @param {string} - string that identifies which flap it is on the fortune teller
+   * @returns {number} - 0 if invalid input, or a number that is associated with the flap
+   */
+  idToNum(flapID) {
+    let number;
+
+    if (this.currentSVG.data.includes('horizontally')) {
+      number = NUMBER_BY_SHADOW_CLICK_REGION.horizontal[flapID];
+    } else if (this.currentSVG.data.includes('vertically')) {
+      number = NUMBER_BY_SHADOW_CLICK_REGION.vertical[flapID];
+    }
+
+    return (number !== undefined) ? number : 0;
+  }
+
+  /**
+   * Activates and handles click logic for the initial closed fortune teller SVG.
+   * @returns {Promise} - Promise object that adds a click event listener to each flap
+   */
+  activateClosedHandler() {
+    return new Promise((resolve) => {
+      this.currentSVG.addEventListener('load', () => {
+        const svgDocument = this.currentSVG.contentDocument;
+        const closedFlaps = svgDocument.querySelectorAll('g[id$="-flap-d"] path[id$="-click"]');
+        closedFlaps.forEach((flap) => {
+          flap.addEventListener('click', (event) => {
+            const flapID = flap.id;
+            const flapColorClicked = this.idToColor(flapID);
+            resolve(flapColorClicked);
+          });
+        });
+      });
+    });
+  }
+
+  /**
+   * Activates and handles click logic for the open fortune teller SVGs.
+   * @returns {Promise} - Promise object that adds a click event listener to each flap
+   */
+  activateNumsHandler() {
+    return new Promise((resolve) => {
+      this.currentSVG.addEventListener('load', () => {
+        const svgDocument = this.currentSVG.contentDocument;
+        const numFlaps = svgDocument.querySelectorAll('path[id$="-click"]');
+        numFlaps.forEach((flap) => {
+          flap.addEventListener('click', (event) => {
+            const flapID = flap.id;
+            const flapNumClicked = this.idToNum(flapID);
+            resolve(flapNumClicked);
+          });
+        });
+      });
+    });
+  }
```

Comment:
these functions are identical aside from the query selector, but I'm fairly sure even that difference could be eliminated. These should be one function. I'm also not sure why this this needs to wait for load and be asynchronous? Why isn't it sufficient to just add the click listeners?